devito/symbolics/printer.py

```diff
@@ -223,3 +251,8 @@ def ccode(expr, **settings):
         the input ``expr`` itself.
     """
     return CodePrinter(settings=settings).doprint(expr, None)
+
+
+# Sympy 1.11 has introduced a bug in `_print_Add`, so we enforce here
+# to always use the correct one from our printer
+setattr(sympy.printing.str.StrPrinter, '_print_Add', CodePrinter._print_Add)
```

Comment:
@FabioLuporini quite hacky but the only "simple" way to have `str` work properly without too much issues while avoiding skipping a sympy version